--- a/src/common/ssl.c.old	2018-08-29 22:13:25.000000000 +0100
+++ b/src/common/ssl.c	2018-11-15 00:58:05.517518299 +0000
@@ -48,6 +48,11 @@
 #define SSL_OP_NO_COMPRESSION 0
 #endif
 
+#if OPENSSL_API_COMPAT >= 0x10100000L
+#define X509_get_notBefore X509_getm_notBefore
+#define X509_get_notAfter X509_getm_notAfter
+#endif
+
 /* globals */
 static struct chiper_info chiper_info;		/* static buffer for _SSL_get_cipher_info() */
 static char err_buf[256];			/* generic error buffer */
@@ -84,8 +89,12 @@
 {
 	SSL_CTX *ctx;
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	SSLeay_add_ssl_algorithms ();
 	SSL_load_error_strings ();
+#else
+	OPENSSL_init_ssl(OPENSSL_INIT_LOAD_SSL_STRINGS, NULL);
+#endif
 	ctx = SSL_CTX_new (SSLv23_client_method ());
 
 	SSL_CTX_set_session_cache_mode (ctx, SSL_SESS_CACHE_BOTH);
@@ -355,8 +364,10 @@
 	ERR_remove_thread_state (NULL);
 #endif
 #else
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
 	ERR_remove_state (0);
 #endif
+#endif
 }
 
 /* Hostname validation code based on OpenBSD's libtls. */
