--- mysql.old/plugin/auth_gssapi/gssapi_server.cc	2016-02-04 17:55:07.116539691 +0000
+++ mysql/plugin/auth_gssapi/gssapi_server.cc	2016-02-04 18:08:14.455317921 +0000
@@ -58,7 +58,7 @@
   }
 
   /* Check for entry in keytab */
-  if (krb5_kt_read_service_key(context, NULL, principal, 0, 0, &key))
+  if (krb5_kt_read_service_key(context, NULL, principal, 0, (krb5_enctype) 0, &key))
   {
     sql_print_warning("GSSAPI plugin : default principal '%s' not found in keytab", unparsed_name);
     goto cleanup;
