--- include/libtorrent/pe_crypto.hpp.old	2015-07-06 12:37:10.672812503 +0100
+++ include/libtorrent/pe_crypto.hpp	2015-07-06 12:41:59.779776529 +0100
@@ -39,7 +39,7 @@
 
 #ifdef TORRENT_USE_GCRYPT
 #include <gcrypt.h>
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 #include <openssl/rc4.h>
 #else
 // RC4 state from libtomcrypt
@@ -115,7 +115,7 @@
 			gcry_cipher_close(m_rc4_incoming);
 			gcry_cipher_open(&m_rc4_incoming, GCRY_CIPHER_ARCFOUR, GCRY_CIPHER_MODE_STREAM, 0);
 			gcry_cipher_setkey(m_rc4_incoming, key, len);
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 			RC4_set_key(&m_remote_key, len, key);
 #else
 			rc4_init(key, len, &m_rc4_incoming);
@@ -132,7 +132,7 @@
 			gcry_cipher_close(m_rc4_outgoing);
 			gcry_cipher_open(&m_rc4_outgoing, GCRY_CIPHER_ARCFOUR, GCRY_CIPHER_MODE_STREAM, 0);
 			gcry_cipher_setkey(m_rc4_outgoing, key, len);
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 			RC4_set_key(&m_local_key, len, key);
 #else
 			rc4_init(key, len, &m_rc4_outgoing);
@@ -159,7 +159,7 @@
 
 #ifdef TORRENT_USE_GCRYPT
 			gcry_cipher_encrypt(m_rc4_outgoing, pos, len, 0, 0);
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 			RC4(&m_local_key, len, (const unsigned char*)pos, (unsigned char*)pos);
 #else
 			rc4_encrypt((unsigned char*)pos, len, &m_rc4_outgoing);
@@ -175,7 +175,7 @@
 
 #ifdef TORRENT_USE_GCRYPT
 			gcry_cipher_decrypt(m_rc4_incoming, pos, len, 0, 0);
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 			RC4(&m_remote_key, len, (const unsigned char*)pos, (unsigned char*)pos);
 #else
 			rc4_encrypt((unsigned char*)pos, len, &m_rc4_incoming);
@@ -186,7 +186,7 @@
 #ifdef TORRENT_USE_GCRYPT
 		gcry_cipher_hd_t m_rc4_incoming;
 		gcry_cipher_hd_t m_rc4_outgoing;
-#elif defined TORRENT_USE_OPENSSL
+#elif ( defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4 )
 		RC4_KEY m_local_key; // Key to encrypt outgoing data
 		RC4_KEY m_remote_key; // Key to decrypt incoming data
 #else
--- src/pe_crypto.cpp~	2014-11-07 03:02:39.000000000 +0000
+++ src/pe_crypto.cpp	2015-07-06 13:51:36.247327809 +0100
@@ -303,7 +303,7 @@
 
 } // namespace libtorrent
 
-#if !defined TORRENT_USE_OPENSSL && !defined TORRENT_USE_GCRYPT
+#if !(defined TORRENT_USE_OPENSSL && !defined OPENSSL_NO_RC4) && !defined TORRENT_USE_GCRYPT
 
 // All this code is based on libTomCrypt (http://www.libtomcrypt.com/)
 // this library is public domain and has been specially
