--- a/ftp/sslapp.c.old	2018-12-18 03:54:18.236119923 +0000
+++ b/ftp/sslapp.c	2018-12-21 00:53:07.225832935 +0000
@@ -65,6 +65,11 @@
   /* init things so we will get meaningful error messages
    * rather than numbers 
    */
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
+  OPENSSL_init_ssl(OPENSSL_INIT_LOAD_SSL_STRINGS |
+		   OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
+  ssl_ctx = SSL_CTX_new(TLS_method());
+#else
   SSL_load_error_strings();
 
   SSLeay_add_ssl_algorithms();
@@ -92,6 +97,7 @@
 	  BIO_printf(bio_err,"Assigned temp (512 bit) RSA key\r\n");
     }
   }
+#endif
 
   /* also switch on all the interoperability and bug
    * workarounds so that we will communicate with people
--- a/ftp/sslapp.h.old	2018-12-18 03:54:18.556114924 +0000
+++ b/ftp/sslapp.h	2018-12-21 00:42:50.219494680 +0000
@@ -30,6 +30,7 @@
 #undef OLDPROTO
 #include <openssl/buffer.h>
 
+#include <openssl/rsa.h>
 #include <openssl/x509.h>
 #include <openssl/ssl.h>
 #include <openssl/err.h>
