diff -Nru openssh-7.3p1.old/cipher-3des1.c openssh-7.3p1/cipher-3des1.c
--- openssh-7.3p1.old/cipher-3des1.c	2016-07-27 23:54:27.000000000 +0100
+++ openssh-7.3p1/cipher-3des1.c	2016-09-09 05:28:22.762537173 +0100
@@ -26,6 +26,8 @@
 
 #include "ssherr.h"
 
+#ifndef OPENSSL_NO_DES
+
 /*
  * This is used by SSH1:
  *
@@ -153,3 +155,5 @@
 	ssh1_3des.flags = EVP_CIPH_CBC_MODE | EVP_CIPH_VARIABLE_LENGTH;
 	return &ssh1_3des;
 }
+
+#endif
diff -Nru openssh-7.3p1.old/cipher.c openssh-7.3p1/cipher.c
--- openssh-7.3p1.old/cipher.c	2016-07-27 23:54:27.000000000 +0100
+++ openssh-7.3p1/cipher.c	2016-09-09 05:30:02.232925284 +0100
@@ -53,9 +53,11 @@
 
 #ifdef WITH_SSH1
 extern const EVP_CIPHER *evp_ssh1_bf(void);
+#ifndef OPENSSL_NO_DES
 extern const EVP_CIPHER *evp_ssh1_3des(void);
 extern int ssh1_3des_iv(EVP_CIPHER_CTX *, int, u_char *, int);
 #endif
+#endif
 
 struct sshcipher {
 	char	*name;
@@ -87,7 +89,9 @@
 #endif /* WITH_SSH1 */
 #ifdef WITH_OPENSSL
 	{ "none",	SSH_CIPHER_NONE, 8, 0, 0, 0, 0, 0, EVP_enc_null },
+#ifndef OPENSSL_NO_DES
 	{ "3des-cbc",	SSH_CIPHER_SSH2, 8, 24, 0, 0, 0, 1, EVP_des_ede3_cbc },
+#endif
 # ifndef OPENSSL_NO_BF
 	{ "blowfish-cbc",
 			SSH_CIPHER_SSH2, 8, 16, 0, 0, 0, 1, EVP_bf_cbc },
@@ -576,9 +580,11 @@
 		break;
 #endif
 #ifdef WITH_SSH1
+#ifndef OPENSSL_NO_DES
 	case SSH_CIPHER_3DES:
 		return ssh1_3des_iv(&cc->evp, 0, iv, 24);
 #endif
+#endif
 	default:
 		return SSH_ERR_INVALID_ARGUMENT;
 	}
@@ -616,9 +622,11 @@
 		break;
 #endif
 #ifdef WITH_SSH1
+#ifndef OPENSSL_NO_DES
 	case SSH_CIPHER_3DES:
 		return ssh1_3des_iv(&cc->evp, 1, (u_char *)iv, 24);
 #endif
+#endif
 	default:
 		return SSH_ERR_INVALID_ARGUMENT;
 	}
