--- spice-gtk-0.34/src/spice-channel.c.old	2017-09-26 00:31:28.265111830 +0100
+++ spice-gtk-0.34/src/spice-channel.c	2017-09-26 00:43:44.685452844 +0100
@@ -392,8 +392,13 @@
 
     g_type_class_add_private(klass, sizeof(SpiceChannelPrivate));
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000
     SSL_library_init();
     SSL_load_error_strings();
+#else
+    OPENSSL_init_ssl(OPENSSL_INIT_LOAD_SSL_STRINGS |
+                     OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
+#endif
 }
 
 /* ---------------------------------------------------------------- */
