--- a/imap-send.c.old	2018-06-21 18:18:50.000000000 +0100
+++ b/imap-send.c	2018-07-23 17:23:59.076711033 +0100
@@ -284,8 +284,13 @@
 	int ret;
 	X509 *cert;
 
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+	OPENSSL_init_ssl(OPENSSL_INIT_LOAD_SSL_STRINGS |
+			 OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
+#else
 	SSL_library_init();
 	SSL_load_error_strings();
+#endif
 
 	meth = SSLv23_method();
 	if (!meth) {
